### PR TITLE
Custom serialization + user agent

### LINKED ISSUE #73: Upgrade dependency on PowerAuth networking
To fully support PowerAuth mobile SDK 1.7.x, we have to upgrade dependency on PowerAuth Networking to version 1.1.0 once is released. 

Related to #69
Related to wultra/networking-android#19

### LINKED ISSUE #74: Custom UserOperation objects
As described in https://github.com/wultra/enrollment-server/pull/307

and implemented on iOS in 

https://github.com/wultra/mtoken-sdk-ios/pull/94

### PR DESCRIPTION
**Pull request by kober32**
- Resolved #74 (added possibility to pass own GSON builder)
- Resolved #73 (upgraded powerauth networking dependency)
- Added `User-Agent` configuration

### CODE REVIEW

**Comment by kober32** [comment]
Converting to draft - still needs to add QR operation changes as in iOS PR https://github.com/wultra/mtoken-sdk-ios/pull/94

**Review by TomasKypta — APPROVED** [review]
Looks OK

File: library/build.gradle.kts [hunk 1]
```diff
@@ -61,18 +61,18 @@ dependencies {
     // DO NOT UPGRADE ABOVE 3.12.X! Version 3.12 is the last version supporting TLS 1 and 1.1
     // If upgraded, the app will crash on android 4.4
     implementation("com.squareup.okhttp3:okhttp:3.12.13")
-    implementation("com.wultra.android.powerauth:powerauth-networking:1.0.2")
+    implementation("com.wultra.android.powerauth:powerauth-networking:1.1.0")
```
Comment (hvge) [inline library/build.gradle.kts:64]: Bump to `1.1.1`

File: library/build.gradle.kts [hunk 2]
```diff
@@ -61,18 +61,18 @@ dependencies {
...
 
     // Dependencies
-    compileOnly("com.wultra.android.powerauth:powerauth-sdk:1.7.0")
+    compileOnly("com.wultra.android.powerauth:powerauth-sdk:1.7.1")
```
Comment (hvge) [inline library/build.gradle.kts:67]: Bump to `1.7.2`

**Review by hvge — APPROVED** [review]
Looks OK

**Pull request merged**